relentless/optimize/method.py

```diff
@@ -339,13 +446,24 @@ def optimize(self, objective):
         iter_num = 0
         cur_res = objective.compute()
         while not self.has_converged(cur_res) and iter_num < self.max_iter:
+            grad_y = self.scale_grad(cur_res.gradient, self.scale)
```

Comment:
Once `scale` is a `KeyedArray`:
```suggestion
            grad_y = scale*cur_res.gradient
```